hydromt_wflow/wflow.py

```diff
@@ -235,95 +227,190 @@ def setup_basemaps(
 
+        # TODO make seperate workflows.river_manning  method
         # Make N_River map from csv file with mapping between streamorder and N_River value
-        if n_river_mapping is None:
-            fn_map = join(DATADIR, "wflow", "N_river_mapping.csv")
-        else:
-            fn_map = n_river_mapping
         strord = self.staticmaps[self._MAPS["strord"]].copy()
-        df = pd.read_csv(fn_map, index_col=0, sep=",|;", engine="python")
+        df = pd.read_csv(rivman_mapping_fn, index_col=0, sep=",|;", engine="python")
         # max streamorder value above which values get the same N_River value
         max_str = df.index[-2]
-        # if streamroder value larger than max_str, assign last value
+        # if streamorder value larger than max_str, assign last value
         strord = strord.where(strord <= max_str, max_str)
         # handle missing value (last row of csv is mapping of missing values)
         strord = strord.where(strord != strord.raster.nodata, -999)
         strord.raster.set_nodata(-999)
-
-        ds_nriver = landuse(
+        ds_nriver = workflows.landuse(
             da=strord,
             ds_like=self.staticmaps,
-            fn_map=fn_map,
+            fn_map=rivman_mapping_fn,
             logger=self.logger,
         )
-
         self.set_staticmaps(ds_nriver)
+        # update config
+        self.set_config("input.lateral.river.bankfull_elevation", self._MAPS["rivzs"])
+
+        # get rivdph, rivwth
+        # while we still have setup_riverwidth one can skip river_bathymetry here
+        # TODO make river_geom_fn required after removing setup_riverwidth
+        if river_geom_fn is not None:
+            gdf_riv = self.data_catalog.get_geodataframe(
+                river_geom_fn, geom=self.region
+            )
+            # reread model data to get river maps
+            inv_rename = {v: k for k, v in self._MAPS.items() if v in self.staticmaps}
+            ds_riv1 = workflows.river_bathymetry(
+                ds_model=self.staticmaps.rename(inv_rename),
+                gdf_riv=gdf_riv,
+                method=rivdph_method,
+                smooth_len=smooth_len,
+                min_rivdph=min_rivdph,
+                min_rivwth=min_rivwth,
+                logger=self.logger,
+                **kwargs,
+            )
+            rmdict = {k: v for k, v in self._MAPS.items() if k in ds_riv1.data_vars}
+            self.set_staticmaps(ds_riv1.rename(rmdict))
+            # update config
+            self.set_config("input.lateral.river.bankfull_depth", self._MAPS["rivdph"])
 
         self.logger.debug(f"Adding rivers vector to staticgeoms.")
         self.staticgeoms.pop("rivers", None)  # remove old rivers if in staticgeoms
         self.rivers  # add new rivers to staticgeoms
 
+    def setup_floodplains(self, elevtn_map="wflow_dem"):
+        """This component adds a hydrologically adjusted river+floodplain elevation map
+        for the 1D-2D local inertial based routing.
+
```

Comment:
add a note to specify hydrologically adjusted with d4 for land and d8 for rivers cellls in docstring?

Reply:
Updated the docstring with much more info.